tests/group_by_test.py

```diff
@@ -97,11 +97,26 @@ def test_group_by_iter(constructor_eager: ConstructorEager) -> None:
     assert sorted(keys) == sorted(expected_keys)
 
 
-def test_group_by_len(constructor: Constructor) -> None:
-    result = (
-        nw.from_native(constructor(data)).group_by("a").agg(nw.col("b").len()).sort("a")
-    )
-    expected = {"a": [1, 3], "b": [2, 1]}
+@pytest.mark.parametrize(
+    ("expr", "expected"),
+    [
+        (nw.col("b").sum(), {"a": [1, 2], "b": [3, 3]}),
+        (nw.col("b").mean(), {"a": [1, 2], "b": [1.5, 3]}),
+        (nw.col("b").max(), {"a": [1, 2], "b": [2, 3]}),
+        (nw.col("b").min(), {"a": [1, 2], "b": [1, 3]}),
+        (nw.col("b").std(), {"a": [1, 2], "b": [0.707107, None]}),
+        (nw.col("b").len(), {"a": [1, 2], "b": [3, 1]}),
+        (nw.col("b").n_unique(), {"a": [1, 2], "b": [3, 1]}),
+        (nw.col("b").count(), {"a": [1, 2], "b": [2, 1]}),
+    ],
+)
+def test_group_by_depth_1_agg(
+    constructor: Constructor,
+    expr: nw.Expr,
+    expected: dict[str, list[int | float]],
+) -> None:
+    data = {"a": [1, 1, 1, 2], "b": [1, None, 2, 3]}
+    result = nw.from_native(constructor(data)).group_by("a").agg(expr).sort("a")
```

Comment:
Also `median` is failing here for pyarrow 🤔, reason why I've left it out. Somehow, though, we're pointing it out in docs that median's results might slightly differ across backends due to differences in the underlying algorithms...